forge-game/src/main/java/forge/game/ability/effects/PlayEffect.java

```diff
@@ -169,6 +169,7 @@ public void resolve(final SpellAbility sa) {
             // so it gets added to stack
             card.setCopiedPermanent(card);
             card.setToken(true);
+            card.setZone(controller.getZone(ZoneType.None));
```

Comment:
Would it work to just cut out `card.setCopiedPermanent` and `card.setToken` and replace them with `card.setCopiedSpell(true)`? I think PlayEffect is the only place where a card is made into a token while on the stack.

Reply:
yea I'm not super happy with how these params are overlapping, investigating a good path for cleanup might be a worthy project
but just as a note if you didn't notice, I did have to fix this area for that last time:
https://github.com/Card-Forge/forge/pull/5230/commits/b05f9f1